src/analysis/interpreter.jl

```diff
@@ -813,10 +806,14 @@ struct MappingInfo <: CC.CallInfo
 end
 
 function _abstract_eval_invoke_inst(interp::DAEInterpreter, inst::Union{CC.Instruction, Nothing}, @nospecialize(stmt), irsv::IRInterpretationState)
-    mi = stmt.args[1]
+    invokee = stmt.args[1]
     RT = Pair{Any, Tuple{Bool, Bool}}
     good_effects = (true, true)
-    m = mi.def
+    if isa(invokee, Core.CodeInstance)
+        m = invokee.def.def
```

Comment:
Eventually (including in Base) we may need to deal with the fact that going from CodeInstance back to MethodInstance may require some extra care to preserve the implied owner, world bounds, and edges information, and invoke pointer that was previously specified. I don't think we have fully specified what that means yet for reflection queries like this, since now these could be cassette like overlays, with a different behavior and ABI owner than the original MethodInstance implied